MTL-1981 - Document operational procedure for configuring NCNs with bonded HSN connections (#5477)
Cray-HPE/docs-csm: operations/configuration_management/Configure_HSN_NIC_Bonding.md

# Configure HSN NIC Bonding

* [Configure HSN NIC Bonding](#configure-hsn-nic-bonding)
    * [References](#references)
    * [Limitations](#limitations)
    * [Prerequisites](#prerequisites)
    * [Setup](#setup)
        * [Fabric configuration](#fabric-configuration)
        * [Host configuration](#host-configuration)
            * [Prepare VCS branch](#prepare-vcs-branch)
            * [Configure the `csm.ncn.hsn_bonding` Ansible role](#configure-the-csmncnhsn_bonding-ansible-role)
            * [Configure CFS to run the `csm.ncn.hsn_bonding` role](#configure-cfs-to-run-the-csmncnhsn_bonding-role)
    * [Verification](#verification)
    * [Troubleshooting](#troubleshooting)
    * [Additional steps](#additional-steps)

This procedure can be used to create a bonded HSN interface on an NCN worker node. The `csm.ncn.hsn_bonding` Ansible role is
merely an automation of the manual steps outlined in the "How to create a bonded IP host interface with HPE Slingshot" document (See [References](#references))

## References

> The "How to create a bonded IP host interface with HPE Slingshot" document is available from the HPE Support Portal. The other documentation is bundled with the HPE Slingshot software download.

Reviewer: ```suggestion
The _How to create a bonded IP host interface with HPE Slingshot_ document is available from the HPE Support Portal. The other documentation is bundled with the HPE Slingshot software download.
```

> * HPE Slingshot Installation Guide for CSM

Reviewer: ```suggestion
* _HPE Slingshot Installation Guide for CSM_
```

> * HPE Slingshot Administration Guide

Reviewer: ```suggestion
* _HPE Slingshot Administration Guide_
```

* [How to create a bonded IP host interface with HPE Slingshot](https://support.hpe.com/hpesc/public/docDisplay?docId=dp00004881en_us&docLocale=en_US)

## Limitations

> * This procedure is only supported on nodes with HPE Cassini interfaces.

Reviewer: ```suggestion
* This procedure is only supported on nodes with HPE Slingshot 200Gbps interfaces.
```

* More than one bonded interface per NCN worker node is not supported.

## Prerequisites

> The following steps should have occurred before configuring a bonded interface on a NCN worker node.

Reviewer 2: `a NCN` to `an NCN` for consistency in the document.

Author: Fixed

> * Slingshot Fabric Manager is installed and configured.

Reviewer: ```suggestion
* HPE Slingshot Fabric Manager is installed and configured.
```

> * Slingshot Host Software is installed and an image containing it has been deployed to the NCN worker nodes.

Reviewer: ```suggestion
* Slingshot Host Software (SHS) is installed and an image containing it has been deployed to the NCN worker nodes.
```

> * User Services Software is installed and an image containing it has been deployed to the NCN worker nodes.

Reviewer: ```suggestion
* User Services Software (USS) is installed and an image containing it has been deployed to the NCN worker nodes.
```

> * Link Aggregation Groups (LAG) have been created using the Slingshot Fabric Manager.

Reviewer: ```suggestion
* Link Aggregation Groups (LAG) have been created using the HPE Slingshot Fabric Manager.
```

> * Link Aggregation Groups (LAG) have been created using the Slingshot Fabric Manager.

Reviewer 2: Could remove `the` here?

Author: Fixed

    * An IP address and netmask have been provided by the fabric administrator.
    * The bonding mode used for the LAG has been provided by the fabric administrator.

>     * The DMAC used for the LAG has been provided by the fabric administrator (See "How to create a bonded IP host interface with HPE Slingshot") for more information.

Reviewer: ```suggestion
    * The DMAC used for the LAG has been provided by the fabric administrator (See the _How to create a bonded IP host interface with HPE Slingshot_ document) for more information.
```

## Setup

### Fabric configuration

Configuring a LAG using the Slingshot Fabric Manager is beyond the scope of this document (See the Link Aggregation section of the
"HPE Slingshot Installation Guide for CSM" for more information) however the following example configuration is provided for the

> purpose of illustration.

Reviewer: ```suggestion
Configuring a LAG using the HPE Slingshot Fabric Manager is beyond the scope of this document (See the "Link Aggregation" section of the
_HPE Slingshot Installation Guide for CSM_ for more information), however the following example configuration is provided for the
purpose of illustration.
```

   git push --set-upstream origin integration-1.26.0
   ```

#### Configure CFS to run the `csm.ncn.hsn_bonding` role

1. Create a CFS configuration using the committed changes.

   1. Obtain the commit hash and create a configuration template file.

      ```bash
      COMMIT=$(git rev-parse --verify HEAD)
      cat << EOF > hsn-nic-bonding.json
      {
        "layers": [
          {
             "cloneUrl": "https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git",
             "commit": "${COMMIT}",
             "name": "hsn-nic-bonding",
             "playbook": "ncn_hsn_bonding.yml"
          }
        ]
      }
      EOF
      ```

   1. Create a CFS configuration from the template file.

      ```bash
      cray cfs configurations update hsn-nic-bonding --file ./hsn-nic-bonding.json
      ```

      Example output:

      ```text
      lastUpdated = "2024-10-11T11:13:39Z"
      name = "hsn-nic-bonding"
      [[layers]]
      cloneUrl = "https://api-gw-service-nmn.local/vcs/cray/csm-config-management.git"
      commit = "e8de8f5be0ba6576b4102137db821a0da3b28375"
      name = "hsn-nic-bonding"
      playbook = "ncn_hsn_bonding.yml"
      ```

> 1. Create a CFS session to apply the configuration to the node(s).

Reviewer: Please indent everything under this step.

          port priority: 255
          port number: 4
          port state: 63

      Slave Interface: macvlan0
      MII Status: up
      Speed: 200000 Mbps
      Duplex: full
      Link Failure Count: 2
      Permanent HW addr: b2:00:00:00:00:01
      Slave queue ID: 0
      Aggregator ID: 1
      Actor Churn State: none
      Partner Churn State: none
      Actor Churned Count: 0
      Partner Churned Count: 2
      details actor lacp pdu:
          system priority: 65535
          system mac address: b2:00:00:00:00:01
          port key: 31
          port priority: 255
          port number: 2
          port state: 63
      details partner lacp pdu:
          system priority: 32768
          system mac address: 02:00:00:00:00:01
          oper key: 2
          port priority: 255
          port number: 3
          port state: 63
      ```

## Troubleshooting

Refer to [View Configuration Session Logs](./View_Configuration_Session_Logs.md) to troubleshoot why the CFS session failed to complete successfully.

If the underlying HSN interfaces are not up or present refer to the Slingshot documentation listed in [References](#references) to verify the fabric is healthy.

> Troubleshooting Slingshot is beyond the scope of this document.

Reviewer: ```suggestion
Troubleshooting HPE Slingshot is beyond the scope of this document. See the _HPE Slingshot Troubleshooting Guide_ for troubleshooting information.
```

## Additional steps

This procedure performs a one time configuration of the target nodes. The bonded HSN configuration will persist through a reboot of the node but a rebuild of the node
will wipe it.

> In order to persist this configuration through a rebuild of the node, the CFS layer should be added to the CFS configuration used for the NCN Worker nodes.

Reviewer: ```suggestion
In order to persist this configuration through a rebuild of the node, the CFS layer should be added to the CFS configuration used for the NCN worker nodes.
```
